src/Containers/Microsoft.NET.Build.Containers/Tasks/ComputeDotnetBaseImageAndTag.cs

```diff
@@ -0,0 +1,245 @@
+/// </summary>
+public sealed class ComputeDotnetBaseImageAndTag : Microsoft.Build.Utilities.Task
+{
+    // starting in .NET 8, the container tagging scheme started incorporating the
+    // 'channel' (rc/preview) and the channel increment (the numeric value after the channel name)
+    // into the container tags.
+    private const int FirstVersionWithNewTaggingScheme = 8;
+
+    /// <summary>
+    /// When in preview, this influences which preview image tag is used, since previews can have compatibility problems across versions.
+    /// </summary>
+    [Required]
+    public string SdkVersion { get; set; }
+
+    /// <summary>
+    /// Used to determine which `tag` of an image should be used by default.
+    /// </summary>
+    [Required]
+    public string TargetFrameworkVersion { get; set; }
+
+    /// <summary>
+    /// Used to inspect the project to see if it references ASP.Net Core, which causes a change in base image to dotnet/aspnet.
+    /// </summary>
+    [Required]
+    public ITaskItem[] FrameworkReferences { get; set; }
+
+    /// <summary>
+    /// If this is set to linux-ARCH then we use jammy-chiseled for the AOT/Extra/etc decisions.
+    /// If this is set to linux-musl-ARCH then we need to use `alpine` for all containers, and tag on `aot` or `extra` as necessary.
+    /// </summary>
+    [Required]
+    public string TargetRuntimeIdentifier { get; set; }
+
+
+    /// <summary>
+    /// If a project is self-contained then it includes a runtime, and so the runtime-deps image should be used.
+    /// </summary>
+    public bool IsSelfContained { get; set; }
+
+    /// <summary>
+    /// If a project is AOT-published then not only is it self-contained, but it can also remove some other deps - we can use the dotnet/nightly/runtime-deps variant here aot
+    /// </summary>
+    public bool IsAotPublished { get; set; }
+
+    /// <summary>
+    /// If the project is AOT'd the aot image variant doesn't contain ICU and TZData, so we use this flag to see if we need to use the `-extra` variant that does contain those packages.
+    /// </summary>
+    public bool UsesInvariantGlobalization { get; set; }
+
+    /// <summary>
+    /// If set, this expresses a preference for a variant of the container image that we infer for a project.
+    /// e.g. 'alpine', or 'jammy-chiseled'
+    /// </summary>
+    public string ContainerFamily { get; set; }
+
+    /// <summary>
+    ///  The final base image computed from the inputs (or explicitly set by the user if IsUsingMicrosoftDefaultImages is true)
+    /// </summary>
+    [Output]
+    public string? ComputedContainerBaseImage { get; private set; }
+
+    private bool IsAspNetCoreProject => FrameworkReferences.Length > 0 && FrameworkReferences.Any(x => x.ItemSpec.Equals("Microsoft.AspnetCore.App", StringComparison.Ordinal));
+    private bool IsMuslRID => TargetRuntimeIdentifier.StartsWith("linux-musl", StringComparison.Ordinal);
```

Comment:
nit: I think this should be named `IsMuslRid`

Reply:
great point, renaming now!